torchgeo/datasets/quakeset.py

```diff
@@ -0,0 +1,289 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""QuakeSet dataset."""
+
+import os
+from collections.abc import Callable
+from typing import Any, cast
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from matplotlib.figure import Figure
+from torch import Tensor
+
+from .geo import NonGeoDataset
+from .utils import DatasetNotFoundError, download_url, percentile_normalization
+
+
+class QuakeSet(NonGeoDataset):
+    """QuakeSet dataset.
+
+    `QuakeSet <https://huggingface.co/datasets/DarthReca/quakeset>`__
+    is a dataset for Earthquake Change Detection and Magnitude Estimation and is used
+    for the Seismic Monitoring and Analysis (SMAC) ECML-PKDD 2024 Discovery Challenge.
+
+    Dataset features:
+
+    * Sentinel-1 SAR imagery
+    * before/pre/post imagery of areas affected by earthquakes
+    * 2 multispectral bands (VV/VH)
+    * 3,327 pairs of pre and post images with 5 m per pixel resolution (512x512 px)
+    * 2 classification labels (unaffected / affected by earthquake)
+    * earthquake magnitudes for each sample
```

Comment:
I guess overall question, do we want to have Classification and Regression supported out of the box?

Reply:
The magnitude is returned by default? But maybe we could also test it with a RegressionTask then?

Reply:
I think the issue is that they both can't be called "label" in the sample dict. So the only workaround would be to support only one or the other

Reply:
This is an interesting observation, will add a note in #985

Reply:
I think we should leave it as a classification task for now and just return the magnitude as additional data if the user wants to use it.